src/lib.rs

```diff
@@ -503,6 +503,25 @@ impl GLRenderer
             .get_webgl2_context(viewport_size_pixels)
     }
 
+    /// Creates a `GLRenderer` from provided [web_sys:: WebGl2RenderingContext].
+    ///
+    /// The parameter `viewport_size_pixels` should be set to
+    /// the initial canvas size, however this can be changed later using
+    /// [GLRenderer:: set_viewport_size_pixels()].
+    #[cfg(target_arch = "wasm32")]
```

Comment:
I hope this is Ok. I am not sure why this does not compile with `doc` and `doctest` also, bc `web-sys` dep has this options in `cargo.toml`.

Reply:
(or, just use the full namespace `web_sys::WebGl2RenderingContext` directly as you've done now, and re-add `doc` and `doctest` to the function `cfg`)